src/unity/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -507,18 +529,87 @@ void activity_classifier::perform_training_iteration() {
   float average_batch_accuracy = cumulative_batch_accuracy / num_batches;
 
   // Report progress if we have an active table printer.
-  // TODO: Report validation metrics.
-  
-  if (training_table_printer_) {
-    training_table_printer_->print_progress_row(
-        iteration_idx, iteration_idx + 1, average_batch_accuracy, average_batch_loss, progress_time());
+  if (validation_data_iterator_ != nullptr) {
+
+    float cumulative_val_accuracy = 0.f;
+    float cumulative_val_loss = 0.f;
```

Comment:
Could we factor out the computation of `cumulative_val_accuracy` and `cumulative_val_loss` into a (single) separate (private) helper method?